syntax = "proto3";

import "eigen.proto";

message SamplerParams {
  message NormalGammaParams {
    double mu0 = 1;
    double a = 2;
    double b = 3;
    double lam_ = 4;
  }

  message BetaParams {
    double a = 1;
    double b = 2;
  }

// Message name is now misleading but the API is safe
  message InverseWishartParams {
    bool is_invgamma = 1;
    double alpha = 2;
    double beta = 3;
    double nu = 4;  // used in case we have a true InvWishart.
    bool identity = 5;
    EigenMatrix V = 6; // Used only if identity is False
  }

  int32 num_components = 1;
  NormalGammaParams p0_params = 2;
  BetaParams rho_params = 3;
  InverseWishartParams sigma_params = 4;

  double mtilde_sigmasq = 5;
}
